packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -281,6 +282,9 @@ export class DataStores implements IDisposable {
         const context = this.contexts.get(envelope.address);
         assert(!!context, 0x162 /* "There should be a store context for the op" */);
         context.process(transformed, local, localMessageMetadata);
+
+        // Notify that a data store changed. This is used to detect if a deleted data store is being used.
+        this.dataStoreChanged(envelope.address);
```

Comment:
This is probably a question not for this PR, but: how this will work in future for nested data stores?
For completely unrelated reason, I though it would be great instead of today's "wrapping" message content from layer to layer instead have a property "path" (either on op itself, or on content) that fully tells where op should go (keeping existing op processing / passing logic as as). This might be useful here maybe?

Reply:
Yeah, something like that will definitely help here.
Couple of thoughts here:
1. This is only temporary while we are building confidence that we are deleting things at a proper cadence. By the time we have nested data stores, we won't need this anymore.
2. Even if we do keep something like this around, there are ways we can do it for nested nodes (DDS and nested data stores): 
  2.1. By doing something like you mentioned where we have the entire path to work with.
  2.2. We can keep peeling the wrappers until we get to the leaf node. Not ideal but since it's temporary it should be fine.
  2.3. By adding callbacks to the summarizer nodes which can eventually call into the root.